### DISCUSSION TITLE
Compile into an exe file

### DISCUSSION THREAD

**Discussion by TianMaiChengGhostRidder** [gleam-lang/gleam#1772] [Ideas & suggestions]
Very good programming language, but how can I compile it into an exe executable? Just like the windows exe file

**Comment by inoas** [comment]
Gleam compiles to either Erlang or JavaScript. Both do not execute without a runtime/vm. For Erlang you will need the BEAM and for JavaScript you will either need a browser or something like Deno or NodeJS.

Once you are very familiar with Gleam and somewhat with Erlang you can try to bundle, for instance with `Burrito`:
https://github.com/burrito-elixir/burrito
Burrito will create a bundle that contains the vm for the specific platform/s.

... or if you don't require the BEAM's cool features, maybe via something like `nexe`
https://www.section.io/engineering-education/compile-your-nodejs-application-into-a-exe-file/ or maybe this https://deno.land/manual@v1.25.4/tools/compiler (I am not familiar with deno)

**Comment by rockwotj** [comment]
Tracking of a native backend for gleam is here: https://github.com/gleam-lang/gleam/issues/109

**Comment by sz3lbi** [comment]
Now, there are also [Node.js single executable applications](https://nodejs.org/api/single-executable-applications.html).
